### DISCUSSION TITLE
Audacious on Windows and certificate validation issues

### DISCUSSION THREAD

**Discussion by AbraxasDMG** [audacious-media-player/audacious#119] [General]
Hi everyone.

I saw an old issue (#1108) on redmine related to this problem.

Due to inability to validate certificate issuer (or other validation reasons), Audacious is unable to play online streams via https. In cases when the http alternative exists, everything is ok. However, there are a lot of online audio sources that do not provide their media through both, http and https. In such situation, the user is unable to consume the stream.

Would it be possible to allow Windows user to activate an alternative option (it can be set directly in %LOCALAPPDATA%\audacious\config, it doesn't have to have a checkbox in the settings window) to ignore validation issues until a better solution comes?

With a limited knowledge I have, I'd assume that it would be enough to allow setting variable like "validate_certificates" in %LOCALAPPDATA%\audacious\config and then update the neon_vfs_verify_environment_ssl_certs function (cert_verification.cc in neon plugin src directory) to check at the beginning if this variable is false. If it is, return fake "successful" result immediately (I guess, something like "failures & ~NE_SSL_UNTRUSTED" should suffice).

Because I am unable to set up development environment to modify and rebuild the neon plugin myself, I tested this by patching the neon.dll and I can confirm that it works perfectly.

Thanks in advance for your consideration.
Cheers!

**Comment by radioactiveman** [comment]
A better approach would be making use of the root certificates. It looks like this has to be done manually on Windows.
In the past LibreOffice has patched neon like this: https://github.com/LibreOffice/core/commit/07073a198a644d67579a00984aae208c79c4f92f 

Maybe the code can be adopted for the neon plugin in Audacious? Are you able to give this a try?

https://github.com/audacious-media-player/audacious-plugins/blob/b4e360a7ed9419a6d2ddf6ed567be5e9cfd5ac0d/src/neon/neon.cc#L633-L638

**Reply by jlindgren90** [reply]
I would love to adopt that patch for our neon plugin on Windows. It looks very elegant. Nice find!

**Reply by radioactiveman** [reply]
Is that still on your radar, @jlindgren90?

**Reply by jlindgren90** [reply]
I have not had time to look at it yet. Getting a Qt 6 build going on Windows will probably come first.

**Reply by jlindgren90** [reply]
https://github.com/audacious-media-player/audacious-plugins/commit/11f9b9d74f58b5cdded1b1a67cadb78f18555fdd is working for me. I ran out of time to figure out the meson.build syntax though -- it needs to link to -lcrypt32.

**Reply by radioactiveman** [reply]
Great news, thanks. :+1: The Meson build works now with https://github.com/audacious-media-player/audacious-plugins/commit/d71c9e6a0643ba02d04e31b90e00a3ac3d7849dd.

**Comment by AbraxasDMG** [comment]
If someone would be able to build the dll and share it with me for testing, I'd be able to provide you feedback. Unfortunately, I wasn't able to figure out myself how to create a build environment that would generate windows binaries.
